src/components/structures/RoomView.js

```diff
@@ -375,6 +376,21 @@ module.exports = React.createClass({
         }
     },
 
+    onSearchContextFillRequest: function(backwards) {
+        if (!this.state.searchContext.getPaginateToken(backwards)) {
+            debuglog("no more context results");
+            return;
+        }
+
+        debuglog("requesting more search context; backwards:"+backwards);
+
+        return MatrixClientPeg.get().paginateEventContext(
+                this.state.searchContext, {backwards: backwards}
+        ).then(() => {
+            this.forceUpdate();
```

Comment:
I really don't like how unintuitive this is. Please add a comment explaining why force updating after this works (because `paginateEventContext` side-effects by adding more events to the `searchContext` passed in.)